### PR TITLE
Fix of multiple bugs in the TPTP output and add basic edge cases tests

### PR DESCRIPTION
**Pull request by jcailler**
Fix TPTP outputs for GDV compatibility: 
* Add `-silent` mode that removes outputs
* Skolem symbols are now `Fun` and not `Id`

### Test-suite update

* Added missing bug files from #27 
* Added basic test suite for simple edge cases 
* Added basic test for TPTP output / SC-TPTP output

### CODE REVIEW

**Comment by jrosain** [comment]
Argh, seems like #29 didn't work. I'll if I can figure out the solution when I have some free time.

File: src/options.go [hunk 1]
```diff
@@ -114,6 +114,14 @@ func buildOptions() {
 			Glob.EnableDebug()
 		},
 		func(bool) {})
+	(&option[bool]{}).init(
+			"silent",
+			false,
+			"Remove logs in the terminal",
```
Comment (jrosain) [inline src/options.go:120]: Actually, not only in the terminal. It avoids all logs (even in files).

File: src/Mods/tptp/proof.go [hunk 2]
```diff
@@ -130,14 +129,14 @@ func makeStep(proof *gs3.GS3Sequent, hypotheses *AST.FormList, new_current_id in
...
+
 			switch initial_formula := proof.GetTargetForm().(type) {
 			case AST.Pred:
-				target2 = get(AST.MakerNot(initial_formula), hypotheses)
-				targetNotNeg = target2
+				targetPos = target
 			case AST.Not:
 				target2 = get(initial_formula.GetForm(), hypotheses)
-				targetNotNeg = target
+				targetPos = target2
```
[the remark this answers is not in the document]
Comment (jrosain) [reply src/Mods/tptp/proof.go:139]: I think I understand the code now. The variable `target2` seems useless and confusing.

**Comment by jrosain** [comment]
Only minor comments, except on the code of closure for TPTP proofs which could be way clearer.

**Comment by jrosain** [comment]
Don't worry about the failures of the test suite, they are unrelated (at least when I'm posting this comment).

File: devtools/test-suite/bugs/bug_27-1.p [hunk 3]
```diff
@@ -0,0 +1,6 @@
+% args: 
+% result: VALID
+
+fof(test,conjecture,
+    a => a).
+
```
Comment (jrosain) [inline devtools/test-suite/bugs/bug_27-1.p:6]: I'm not sure what this test is supposed to do.
It runs without problem on the version on master.
What did you want to test?

File: devtools/test-suite/bugs/bug_27-2.p [hunk 4]
```diff
@@ -0,0 +1,8 @@
+% args: 
+% result: VALID
+
+fof(an, axiom, 
+    p).
+
+fof(the, conjecture, 
+    p).
```
Comment (jrosain) [inline devtools/test-suite/bugs/bug_27-2.p:8]: Same comment.

File: devtools/test-suite/bugs/bug_27-3.p [hunk 5]
```diff
@@ -0,0 +1,10 @@
+% args: 
+% result: VALID
+
+fof(an, axiom, 
+    ? [X] :
+    p(X)).
+
+fof(the, conjecture, 
+    ? [X] :
+    p(X)).
```
Comment (jrosain) [inline devtools/test-suite/bugs/bug_27-3.p:10]: Again, same comment.

File: src/Glob/helper.go [hunk 6]
```diff
@@ -249,11 +261,23 @@ func IsPreInnerSko() bool {
...
 }
 
+func GetVersion() string {
+	return version
+}
+
+func GetCommit() string {
+	return commit
+}
+
```
Comment (jrosain) [inline src/Glob/helper.go:271]: I'm for a more local management of version.
Instead of returning the commit hash, I'd return it inside the version `v1.2-dev.r{commit}` (r stands for revision).
What do you think?

File: src/main.go [hunk 7]
```diff
@@ -112,6 +112,10 @@ func presearchLoader() (AST.Form, int) {
 
 	problem := os.Args[len(os.Args)-1]
 	Glob.SetProblemName(path.Base(problem))
+
+	fmt.Printf("You are running Goeland v.%v --- commit: %v\n", Glob.GetVersion(), Glob.GetCommit())
```
Comment (jrosain) [inline src/main.go:116]: c.f. my comment on the versioning

**Comment by jrosain** [comment]
Almost all good to me! Some comments on the new things introduced by your last commit.
I'll launch the full ci now as I don't think the requested changes will change the result.

**Review by jrosain — CHANGES_REQUESTED** [review]

File: src/build-hash
Comment (jrosain) [inline src/build-hash:1]: Remove this file please

File: devtools/test-suite/bugs/bug_27-1.p (hunk 3 above)
Comment (jrosain) [reply devtools/test-suite/bugs/bug_27-1.p:6]: Solved as retrospective tests for a previous PR.

**Comment by jrosain** [comment]
Seems to be working (never failed in local even with multiple launches).

I think that we should include a `-version` option, and after a squash it will be ready to merge I think.
